draft-ietf-tls-esni.md

```diff
@@ -837,11 +837,16 @@ If the server provided "retry_configs" and if at least one of the values
 contains a version supported by the client, the client can regard the ECH keys
 as securely replaced by the server. It SHOULD retry the handshake with a new
 transport connection, using the retry configurations supplied by the
-server. The retry configurations may only be applied to the retry
-connection. The client MUST NOT use retry configurations for connections beyond
-the retry. This avoids introducing pinning concerns or a tracking vector,
-should a malicious server present client-specific retry configurations in order
-to identify the client in a subsequent ECH handshake.
+server. (Clients can implement a new transport connection in a way that best
+suits their deployment. For example, clients can reuse the same IP address
+when establishing the new transport connection or they can choose to use a
+different IP address if provided with options from DNS. ECH does not mandate
+any specific implementation choices when establishing this new connection.)
+The retry configurations may only be applied to the retry connection. The
+client MUST NOT use retry configurations for connections beyond the retry.
```

Comment:
Unrelated to this change, but if a client is making multiple connections (hello HTTP/1.1), can it use the retry configuration it learns from its first connection to guide all connections in that pool?

Let's say that the client is making these connections within the same context and so has no concerns about the connection attempts being correlated (maybe it was going to send the same cookies in all of them).  Should the client only use the retry configuration for a single connection, even knowing that it will be forced to retry multiple times, or can it propagate that information immediately?

I like that this text largely avoids 2119 language (questionable lowercase "may" and "should" usage notwithstanding), but this one comes with a "MUST NOT" that seems misguided.

Reply:
Indeed -- unrelated, since this is existing text. I filed [this issue](https://github.com/tlswg/draft-ietf-tls-esni/issues/565) to track resolution separately.